lib/inets/src/http_client/httpc.erl

```diff
@@ -75,85 +193,190 @@ profile_name(Profile) ->
+%%     Default is `none'.</dd>
+%%   <dt>`body_format'</dt>
+%%   <dd>Defines if the body is to be delivered as a `string' or `binary'. This
+%%   option is only valid for the synchronous request.
+%%
+%%     Default is `string'.</dd>
+%%   <dt>`full_result'</dt>
+%%   <dd> Defines if a "full result" is to be returned to the caller (that is,
+%%   the body, the headers, and the entire status line) or not (the body and the
+%%   status code).
+%%
+%%     Default is `true'.</dd>
+%%   <dt>`headers_as_is'</dt>
+%%   <dd> Defines if the headers provided by the user are to be made lower case or
+%%     to be regarded as case sensitive.
+%%
+%%     The HTTP standard requires them to be case insensitive. Use this feature
+%%     only if there is no other way to communicate with the server or for
+%%     testing purpose. When this option is used, no headers are automatically
+%%     added. All necessary headers must be provided by the user.
+%%
+%%     Default is `false'.</dd>
+%%   <dt>`socket_opts'</dt>
+%%   <dd> Socket options to be used for this request.
+%%
+%%     Overrides any value set by function {@link set_options/1} {@link set_option/2}.
+%%
+%%     The validity of the options is not checked by the HTTP client they are
+%%     assumed to be correct and passed on to `ssl' application and inet driver,
+%%     which may reject them if they are not correct.
+%%
+%%     <note>
+%%     Persistent connections are not supported when setting the socket_opts
+%%     option. When socket_opts is not set the current implementation assumes
+%%     the requests to the same host, port combination will use the same socket
+%%     options.
+%%     </note>
+%%
+%%     By default the socket options set by functions {@link set_options/1}
+%%     {@link set_options/2} are used when establishing a connection.</dd>
+%%   <dt>`receiver'</dt>
+%%   <dd> Defines how the client delivers the result of an asynchronous request
+%%     (`sync' has the value `false'). The `receiver' (type {@link receiver()})
+%%     <dl>
+%%     <dt>`pid()'</dt>
+%%     <dd>Messages are sent to this process in the format `{http, ReplyInfo}'</dd>
+%%     <dt>`function/1'</dt>
+%%     <dd>Information is delivered to the receiver through calls to the provided `fun Receiver(ReplyInfo)'.</dd>
+%%     <dt>`{Module, Function, Args}'</dt>
+%%     <dd>Information is delivered to the receiver through calls to the
+%%       callback function `apply(Module, Function, [ReplyInfo | Args])'.
+%%       In all of these cases, `ReplyInfo' has the following structure:
+%%       ```
+%%       {RequestId, saved_to_file}
+%%       {RequestId, {error, Reason}}
+%%       {RequestId, Result}
+%%       {RequestId, stream_start, Headers}
+%%       {RequestId, stream_start, Headers, HandlerPid}
+%%       {RequestId, stream, BinBodyPart}
+%%       {RequestId, stream_end, Headers}
+%%       '''
+%%
+%%       Default is the pid of the process calling the request function (self()).</dd>
+%%     </dl>
+%%   </dd>
+%% </dl>
```

Comment:
This chunk of edoc style documentation is new isn't it? But it is redundant as long as we have the same text in the `httpc.xml` file. So if there is no additions here compared with in the xml file I suggest that we skip documentation in the .erl file.

Reply:
Got it. I agree with you for this wall of text. However, I am not sure that we should remove minimal (one-liners) documentation from source code. My reasoning is that wall-of-text belongs to XML files, but short and concise info may stay in *.erl files, even if creating a minor duplication because it helps us, developers, understand what the function should do without navigating to another file.

- Do you think we should remove all documentation from source code if it is already present in XML?

Reply:
I agree that we should have short few line descriptions of each function and also other comments explaining what's going on in the code. But not the complete documentation intended for the users of the API.